### ISSUE TITLE
[feature][@angular/elements] Expose ComponentNgElementStrategy

### ISSUE THREAD

**Issue by flash-me** [angular/angular#33490]
# 🚀 feature request

### Relevant Package
@angular/elements

### Description
There is no "easy" way to pass custom [lifecycle callbacks](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-reactions) for CustomElements.
We may pass a custom **NgElementStrategyFactory** to `createCustomElement`, but when looking at [ComponentNgElementStrategy](https://github.com/angular/angular/blob/master/packages/elements/src/component-factory-strategy.ts#L45), this is not something that should be done by the average angular developer.

### Describe the solution you'd like
1. Suggestion: You may expose **ComponentNgElementStrategy**. That way we could extend it and override the methods + call the super methods.
2. Suggestion: Extend `createCustomElement` to directly pass the **optional** custom callbacks, which in turn are called internally. Maybe an object of: 
```ts
export interface CustomElementCallbacks {
  /** When upgraded, its constructor is run, with no arguments */
  constructedCallback()?: void;
  /** When it becomes connected, its connectedCallback is called, with no arguments */
  connectedCallback()?: void;
  /** When it becomes disconnected, its disconnectedCallback is called, with no arguments */
  disconnectedCallback()?: void;
  /** When it is adopted into a new document, its adoptedCallback is called, given the old document and new document as arguments */
  adoptedCallback()?: void;
  /** 
  * When any of its attributes are changed, appended, removed, or replaced, 
  * its attributeChangedCallback is called, given 
  * the attribute's local name, old value, new value 
  */
  attributeChangedCallback(attrName: string, oldVal?: any, newVal?: any)?: void;
}
```
### Describe alternatives you've considered
Just copy the code from [here](https://github.com/angular/angular/blob/master/packages/elements/src/component-factory-strategy.ts)? (Not thinking about compatibility on version changes, though....)

**Comment by remackgeek** [comment]
Exposing ComponentNgElementStrategy is a duplicate of  [#26112](https://github.com/angular/angular/issues/26112)
There is a pull request that was never merged: [#26216](https://github.com/angular/angular/pull/26216)
I would love to see this in v9!

**Comment by flash-me** [comment]
Since the related PR has been closed, I'm closing this, too.
